### ISSUE TITLE
Support aggregation stage names starting with '$'

### ISSUE THREAD

**Issue by RaschidJFR** [parse-community/parse-server#7338]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

In the JS SDK `Parse.Query.aggregate()` works really well in runtime, but the fact that stage names differ from the built-in MongoDB’s by the `$` symbol represents an issue for compatibility, for example, when importing/exporting/debugging a pipeline in other tools like MongoCompass.

### Feature / Enhancement Description

`Parse.Query.aggregate()` should support pipeline stage names in the same formate as in mongodb so pipelines can be copy-pasted from/to other tools (mongo shell, mongo compass, etc), without having to edit each stage's name. 

### Example Use Case

Either of these parameters should be valid:

```js
// current Parse SDK format
Parse.Query.aggregate({ match: { _id: '1234' } });
```

```js
// mongodb's standard format
Parse.Query.aggregate({ $match: { _id: '1234' } });
```

### Alternatives / Workarounds

A current workaround is to develop/test/debug pipelines in MongoDB Compass (or your preferred tool) and then copy the whole pipeline into your js code and edit all stage names (only top-level, btw) by adding a preceding `$`.

**Comment by mtrezza** [comment]
Thanks for suggesting!

I think this change would be welcome by the community. I have stumbled upon this inconvenience too, and it really can be quite a big one in a large pipeline.

The conversion back and forth between MongoDB and Parse pipeline syntax during development is additionally difficult because the IDE doesn't check the syntax, for the IDE it's just a large JSON.

**Comment by mtrezza** [comment]
I just noticed that the example in the issue actually contains 2 changes that are unrelated to each other.

- a) Remove $ sign
- b) Rename `objectId` to `_id`

These be distinctly discussed within this PR, or even be two separate issues with 2 separate PRs if their respective implications differ substantially.

Could you make this distinction?

**Comment by RaschidJFR** [comment]
You're right. I decided to make that `_id` change last minute when I noticed that the method [transformStage()](https://github.com/parse-community/parse-server/blob/master/src/Routers/AggregateRouter.js#L106) is doing it anyway for `$group`. But indeed, I must make sure it works for other stages, for example, `$match`.

I think that changing only the `$` part is just halfway through the solution as it would still be incompatible with a native mongo pipeline. 

Would you agree @mtrezza that I should dig into this to release a complete solution in the PR?

**Comment by mtrezza** [comment]
Doing this in 1 PR combines 2 breaking changes into 1, so they will also deprecate together, which may be easier for the developer to follow and the developer only has to touch the pipeline once.

In any case, these 2 distinct changes have to be clearly reflected in this issue's title and description. Each of the changes requires separate tests and each of them needs to be evaluated separately regarding SDK compatibility.

**Comment by mtrezza** [comment]
@RaschidJFR If you need any guidance on how to proceed please feel free to let me know.

**Comment by mtrezza** [comment]
Closed via https://github.com/parse-community/parse-server/pull/7339
